ndcube/ndcube.py

```diff
@@ -792,3 +801,90 @@ def plot(self, *args, **kwargs):
                 "no default plotting functionality is available.")
 
         return self.plotter.plot(*args, **kwargs)
+
+    def _new_instance_from_op(self, new_data, new_unit):
+        # This implicitly assumes that the arithmetic operation does not alter
+        # the WCS, mask, or uncertainty
+        new_cube = type(self)(new_data,
+                              unit=new_unit,
+                              wcs=self.wcs,
+                              mask=self.mask,
+                              meta=self.meta,
+                              uncertainty=self.uncertainty)
+        if self.extra_coords is not None:
+            new_cube._extra_coords = deepcopy(self.extra_coords)
+        return new_cube
+
+    def __neg__(self):
+        return self._new_instance_from_op(-self.data, self.unit)
+
+    def __pow__(self, value):
+        new_data = self.data ** value
+        new_unit = self.unit ** value
+        return self._new_instance_from_op(new_data, new_unit)
+
+    def __add__(self, value):
+        if hasattr(value, 'unit'):
+            if isinstance(value, u.Quantity):
+                # NOTE: if the cube does not have units, we cannot
+                # perform arithmetic between a unitful quantity.
+                # This forces a conversion to a dimensionless quantity
+                # so that an error is thrown if value is not dimensionless
+                cube_unit = u.Unit('') if self.unit is None else self.unit
+                new_data = self.data + value.to_value(cube_unit)
+            else:
+                # NOTE: This explicitly excludes other NDCube objects and NDData objects
+                # which could carry a different WCS than the NDCube
+                return NotImplemented
+        else:
+            new_data = self.data + value
+        return self._new_instance_from_op(new_data, self.unit)
+
+    def __radd__(self, value):
+        return self.__add__(value)
+
+    def __sub__(self, value):
+        return self.__add__(-value)
+
+    def __rsub__(self, value):
+        return self.__neg__().__add__(value)
+
+    def __mul__(self, value):
+        if hasattr(value, 'unit'):
+            if isinstance(value, u.Quantity):
+                # NOTE: if the cube does not have units, set the unit
+                # to dimensionless such that we can perform arithmetic
+                # between the two.
+                cube_unit = u.Unit('') if self.unit is None else self.unit
+                new_data = self.data * value.to_value()
+                new_unit = cube_unit * value.unit
+            else:
+                return NotImplemented
+        else:
+            new_data = self.data * value
+            new_unit = self.unit
+        return self._new_instance_from_op(new_data, new_unit)
+
+    def __rmul__(self, value):
+        return self.__mul__(value)
+
+    def __truediv__(self, value):
+        return self.__mul__(1/value)
+
+    def __rtruediv__(self, value):
```

Comment:
Remind me why this can't also be ```return self.__truediv__(value)```.  Will all this checking and handling then ultimately be done by ```self.__mul__```

Reply:
`__truediv__` is `cube / value` while `__rtruediv__` is `value/cube` so the implicit assumption in `__mul__` is that `cube` is in the numerator. This is not true here. Actually though, this could be implemented as `__pow__(-1).__mul__(value)__`...

Reply:
Of course you’re right!